### PR TITLE
Fix FHIRPath expression (FHIR-46103)

### PR DESCRIPTION
**Pull request by dbojicic**
Fix for FHIR-46103.

### CODE REVIEW

File: input/resources/au-core-organization.xml [hunk 1]
```diff
@@ -46,7 +46,7 @@
         <key value="au-core-org-01"/>
         <severity value="error"/>
         <human value="A National Organisation Identifier (type=NOI) shall be an HPI-O or PAI-O"/>
-        <expression value="where(type.coding.code = 'NOI').exists() implies (system='http://ns.electronichealth.net.au/id/hi/hpio/1.0') or (system='http://ns.electronichealth.net.au/id/pcehr/paio/1.0')"/>
+        <expression value="type.coding.where(code='NOI').exists() implies (system='http://ns.electronichealth.net.au/id/hi/hpio/1.0') or (system='http://ns.electronichealth.net.au/id/pcehr/paio/1.0')"/>
```
Comment (dt-r) [inline input/resources/au-core-organization.xml:49]: FHIRPath gets rid of the warnings but doesn't behave as desired. Test against this snippet, which should fail but doesn't: 
```
<identifier>
    <type>
      <coding>
        <system value="http://terminology.hl7.org/CodeSystem/v2-0203"/>
        <code value="XX"/>
      </coding>
      <coding>
        <system value="http://terminology.hl7.org.au/CodeSystem/v2-0203"/>
        <code value="NOI"/>
        <display value="National Organisation Identifier"/>
      </coding>
      <text value="ACN"/>
    </type>
    <system value="http://hl7.org.au/id/acn"/>
    <!--  Note: 518247535 is an invalid ACN ie does not pass the ACN check digit algorithm  -->
    <value value="518247535"/>
  </identifier>
```

**Review by dt-r — CHANGES_REQUESTED** [review]
Fix the FHIRPath - it's not iterating across the collection.

File: input/resources/au-core-organization.xml (hunk 1 above)
Comment (dbojicic) [reply input/resources/au-core-organization.xml:49]: Tested by adding above snippet and it failed as expected:  
Constraint failed: au-core-org-01: 'A National Organisation Identifier (type=NOI) shall be an HPI-O or PAI-O' (defined in http://hl7.org.au/fhir/core/StructureDefinition/au-core-organization).

**Comment by brettesler-ext** [comment]
I think the NOI is detected fine for any of the coding but am not sure on the implies / or order of operations so would prefer bracketing the or expression just for clarity 
type.coding.where(code='NOI').exists() implies ((system='http://ns.electronichealth.net.au/id/hi/hpio/1.0') or (system='http://ns.electronichealth.net.au/id/pcehr/paio/1.0'))

**Review by dt-r — APPROVED** [review]

**Pull request merged**